### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by tsalo**
Tagging @handwerkerd for your review.

For the most part, my changes just leverage restructuredText (e.g., by using the `:doc:` role instead of explicit hyperlinks). I think the documentation looks great, although one thing I don't like is splitting off the output filename documentation from the general output documentation. From the perspective of a user, I think being able to look at a list of files and seeing what each means is more immediately useful than a walkthrough of the HTML report, or really anything else on the main outputs page.

### CODE REVIEW

File: docs/building_decision_trees.rst [hunk 1]
```diff
@@ -255,53 +259,59 @@ classified as 'accepted' or 'rejected' by the time the tree is completed.
...
 ********************************
 
 There are several expectations for selection functions that are necessary for them to
-properly execute. In `selection_nodes.py`_, ``manual_classify``, ``dec_left_op_right``,
-and ``calc_kappa_rho_elbows_kundu`` are good examples for how to meet these expectations.
+properly execute.
+In :mod:`~tedana.selection.selection_nodes`,
+:func:`~tedana.selection.selection_nodes.manual_classify`,
+:func:`~tedana.selection.selection_nodes.dec_left_op_right`,
+and :func:`~tedana.selection.selection_nodes.calc_kappa_rho_elbows_kundu`
```
Comment (handwerkerd) [inline docs/building_decision_trees.rst:313]:
```suggestion
and :func:`~tedana.selection.selection_nodes.calc_kappa_elbow`
```

Since I wrote docs, the kappa & rho elbow calcs were split into two functions. Just referencing one here.

**Review by handwerkerd — CHANGES_REQUESTED** [review]
I went through all the new links to the API and found one dead link (my fault since I changed the name of a function).

For the outputs of tedana, outputs.html is the main link in the contents. That page was getting very long and hard to follow so I split off output_file_descriptions.html and classification_output_descriptions.html onto separate pages. I still think we don't want all the information on a single super-long page, but I could see combining those two pages that are linked from the main page. Thoughts?

File: docs/building_decision_trees.rst (hunk 1 above)
Comment (tsalo) [reply docs/building_decision_trees.rst]: Thanks!

**Comment by tsalo** [comment]
@handwerkerd if I understand you correctly (i.e., adding both files to the table of contents, so they show up on the sidebar), then I agree completely. I think that is a great solution.

**Review by handwerkerd — APPROVED** [review]

**Comment by handwerkerd** [comment]
I'm merging this & will then re-org the outputs descriptions.
